src/plugins/liveobjects/liveobjects.ts

```diff
@@ -84,12 +92,21 @@ export class LiveObjects {
   /**
    * @internal
    */
-  handleStateMessages(stateMessages: StateMessage[]): void {
+  handleStateMessages(stateMessages: StateMessage[], msgRegionalTimeserial: string | null | undefined): void {
+    const timeserial = DefaultTimeserial.calculateTimeserial(this._client, msgRegionalTimeserial);
+
     if (this._syncInProgress) {
-      // TODO: handle buffering of state messages during SYNC
+      // the client receives state messages in realtime over the channel concurrently with the SYNC sequence,
+      // which means that some of the state messages may already be applied by the realtime in the SYNC sequence once it's ended.
+      // to avoid re-applying those state messages (for example double counting COUNTER_INC operation),
+      // we buffer incoming state operation messsages while SYNC is in progress, and apply them once SYNC has ended.
```

Comment:
```suggestion
      // The client receives state messages in realtime over the channel concurrently with the SYNC sequence.
	  // Some of the incoming state messages may have already been applied to the state objects described in
	  // the SYNC sequence, but others may not; therefore we must buffer these messages so that we can apply
	  // them to the state objects once the SYNC is complete. To avoid double-counting, the buffered operations
	  // are applied according to the state object's regional timeserial, which reflects the regional timeserial
	  // of the state message that was last applied to that state object.
```